### PR TITLE
AHRS: added AHRS_OPTIONS and jamming simulator

### PR DESCRIPTION
**Pull request by tridge**
this brings in the separable parts of #24899 to try to progress that work

this includes the AHRS_OPTIONS addition to allow disabling of DCM fallback in plane

### CODE REVIEW

File: libraries/AP_AHRS/AP_AHRS.cpp [hunk 1]
```diff
@@ -1930,26 +1927,54 @@ AP_AHRS::EKFType AP_AHRS::_active_EKF_type(void) const
...
-            should_use_gps &&
-            AP::gps().status() >= AP_GPS::GPS_OK_FIX_3D) {
-            // if the EKF is not fusing GPS or doesn't have a 2D fix
-            // and we have a 3D lock, then plane and rover would
-            // prefer to use the GPS position from DCM. This is a
-            // safety net while some issues with the EKF get sorted
-            // out
+
+        // Handle fallback for the case where the DCM or EKF is unable to provide attitude or height data.
+        const bool cant_use_dcm = dcm.yaw_source_available() && !fly_forward;
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_AHRS/AP_AHRS.cpp:1932]: good catch, thanks!

[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_AHRS/AP_AHRS.cpp:1932]: fixed

**Comment by Ryanf55** [comment]
We would like @robertlong13 to test this before merge to make sure it meets his needs.

**Review by robertlong13 — APPROVED** [review]
This is perfect. Exactly what I need. I like that you kept it so that it still falls back if the EKF entirely gives up on attitude or vertical position.

Tested with a quick QHover/FBWB flight in RealFlight.

**Pull request merged**